k3s-io/k3s
# Move the apiserver addresses controller into the etcd package

#### Proposed Changes ####

This controller only needs to run when using managed etcd, so move it in
with the rest of the etcd stuff. This change also modifies the
controller to only watch the Kubernetes service endpoint, instead of
watching all endpoints in the entire cluster.

Fixes an error message revealed by use of a newer grpc client in
Kubernetes 1.24, which logs an error when the Put to etcd failed because
kine doesn't support the etcd Put operation. The controller shouldn't
have been running without etcd in the first place.

#### Types of Changes ####

bugfix

#### Verification ####

* See linked issue
* Confirm that `k3s server` logs do not contain any mention of "managed etcd" when not using managed etcd.

#### Linked Issues ####

* #5389 

#### User-Facing Change ####
```release-note
Managed etcd controllers no longer run on clusters not using managed etcd.
```

#### Further Comments ####
